qiskit/algorithms/optimizers/optimizer.py

```diff
@@ -202,6 +202,43 @@ def optimize(self, num_vars, objective_function, gradient_function=None,
                 self.__class__.__name__)
         pass
 
+    def _validate_optimize_input(
```

Comment:
In the abstract `optimize` method above this code is duplicated, could you make the `optimize` function call `_validate_optimize_input` instead? 🙂

Reply:
At first, I thought I deprecated Optimizer.optimize, but I changed it to use existing one.